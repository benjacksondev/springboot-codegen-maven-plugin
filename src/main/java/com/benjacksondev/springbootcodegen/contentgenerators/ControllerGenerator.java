package com.benjacksondev.springbootcodegen.contentgenerators;

import com.benjacksondev.springbootcodegen.AbstractContentGenerator;
import com.benjacksondev.springbootcodegen.ContentGeneratorResult;
import com.benjacksondev.springbootcodegen.StringExtensions;

public class ControllerGenerator extends AbstractContentGenerator {

    @Override
    public ContentGeneratorResult generate() {
        return new ContentGeneratorResult(getClassName(), getClassContent());
    }

    private String getClassName() {
        return componentName + "Controller";
    }

    private String getClassContent() {
        String serviceName = componentName + "Service";
        String entityName = componentName;

        return "package " + packageName + ";\n\n"
                + "import java.util.List;\n"
                + "import org.springframework.beans.factory.annotation.Autowired;\n"
                + "import org.springframework.stereotype.Controller;\n"
                + "import org.springframework.ui.Model;\n"
                + "import org.springframework.validation.BindingResult;\n"
                + "import org.springframework.web.bind.annotation.GetMapping;\n"
                + "import org.springframework.web.bind.annotation.PostMapping;\n"
                + "import org.springframework.web.bind.annotation.RequestMapping;\n"
                + "import org.springframework.web.bind.annotation.RequestParam;\n"
                + "import jakarta.validation.Valid;\n\n"
                + "@Controller\n"
                + "@RequestMapping(\"/" + entityName.toLowerCase() + "\")\n"
                + "public class " + componentName + "Controller {\n\n"
                + "    @Autowired\n"
                + "    " + serviceName + " " + StringExtensions.studlyCapsToCamelCase(componentName) + ";\n\n"
                + "    @GetMapping(\"\")\n"
                + "    public String showAll(@RequestParam(required = false) String message, Model model) {\n"
                + "        List<" + entityName + "> " + StringExtensions.studlyCapsToCamelCase(entityName) + "s = " + StringExtensions.studlyCapsToCamelCase(entityName) + ".get" + entityName + "s();\n"
                + "        model.addAttribute(\"" + entityName.toLowerCase() + "s\", " + StringExtensions.studlyCapsToCamelCase(entityName) + "s);\n\n"
                + "        return \"" + entityName.toLowerCase() + "/index\";\n"
                + "    }\n\n"
                + "    @PostMapping(\"/create\")\n"
                + "    public String create(@Valid " + entityName + " " + StringExtensions.studlyCapsToCamelCase(entityName) + ", BindingResult result, Model model) {\n"
                + "        if (result.hasErrors()) {\n"
                + "            return \"redirect:/" + entityName.toLowerCase() + "/add\";\n"
                + "        }\n\n"
                + "        " + serviceName.toLowerCase() + ".create" + entityName + "(" + entityName.toLowerCase() + ");\n"
                + "        model.addAttribute(\"message\", \"some message\");\n"
                + "        return \"redirect:/" + entityName.toLowerCase() + "\";\n"
                + "    }\n"
                + "}\n";
    }

}
