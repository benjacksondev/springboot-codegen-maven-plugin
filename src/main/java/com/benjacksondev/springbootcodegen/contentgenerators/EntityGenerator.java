package com.benjacksondev.springbootcodegen.contentgenerators;

import com.benjacksondev.springbootcodegen.AbstractContentGenerator;
import com.benjacksondev.springbootcodegen.ContentGeneratorResult;

public class EntityGenerator extends AbstractContentGenerator {
    
    @Override
    public ContentGeneratorResult generate() {
        return new ContentGeneratorResult(getClassName(), getClassContent());
    }
    
    private String getClassName() {
        return componentName;
    }

    private String getClassContent() {
        return "package " + packageName + ";\n\n"
                + "import jakarta.persistence.Entity;\n"
                + "import jakarta.persistence.GeneratedValue;\n"
                + "import jakarta.persistence.GenerationType;\n"
                + "import jakarta.persistence.Id;\n\n"
                + "@Entity\n"
                + "public class " + componentName + " {\n\n"
                + "    @Id\n"
                + "    @GeneratedValue(strategy = GenerationType.IDENTITY)\n"
                + "    private Long id;\n\n"
                + "    public " + componentName + "() {\n"
                + "    }\n\n"
                + "    public Long getId() {\n"
                + "        return id;\n"
                + "    }\n"
                + "}\n";
    }

}
