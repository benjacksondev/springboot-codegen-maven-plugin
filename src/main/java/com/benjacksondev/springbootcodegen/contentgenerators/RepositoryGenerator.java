package com.benjacksondev.springbootcodegen.contentgenerators;

import com.benjacksondev.springbootcodegen.AbstractContentGenerator;
import com.benjacksondev.springbootcodegen.ContentGeneratorResult;

public class RepositoryGenerator extends AbstractContentGenerator {

    @Override
    public ContentGeneratorResult generate() {
        return new ContentGeneratorResult(getClassName(), getClassContent());
    }

    private String getClassName() {
        return componentName + "Repository";
    }

    private String getClassContent() {
        String entityName = componentName;

        return "package " + packageName + ";\n\n"
                + "import org.springframework.data.repository.CrudRepository;\n\n"
                + "public interface " + componentName + "Repository extends CrudRepository<" + entityName + ", Long> {\n"
                + "}\n";
    }
}
