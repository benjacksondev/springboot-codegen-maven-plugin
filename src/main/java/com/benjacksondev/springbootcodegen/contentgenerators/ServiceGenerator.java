package com.benjacksondev.springbootcodegen.contentgenerators;

import com.benjacksondev.springbootcodegen.AbstractContentGenerator;
import com.benjacksondev.springbootcodegen.ContentGeneratorResult;

public class ServiceGenerator extends AbstractContentGenerator {

    @Override
    public ContentGeneratorResult generate() {
        return new ContentGeneratorResult(getClassName(), getClassContent());
    }

    public String getClassName() {
        return componentName + "Service";
    }

    public String getClassContent() {
        String entityName = componentName;
        String repositoryName = componentName + "Repository";

        return "package " + packageName + ";\n\n"
                + "import java.util.ArrayList;\n"
                + "import java.util.List;\n"
                + "import org.springframework.beans.factory.annotation.Autowired;\n"
                + "import org.springframework.stereotype.Service;\n\n"
                + "@Service\n"
                + "public class " + componentName + "Service {\n\n"
                + "    @Autowired\n"
                + "    " + repositoryName + " " + entityName.toLowerCase() + "Repository;\n\n"
                + "    public " + entityName + " create" + entityName + "(" + entityName + " " + entityName.toLowerCase() + ") {\n"
                + "        return " + entityName.toLowerCase() + "Repository.save(" + entityName.toLowerCase() + ");\n"
                + "    }\n\n"
                + "    public List<" + entityName + "> get" + entityName + "s() {\n"
                + "        List<" + entityName + "> " + entityName.toLowerCase() + "List = new ArrayList<>();\n"
                + "        " + entityName.toLowerCase() + "Repository.findAll().forEach(" + entityName.toLowerCase() + "List::add);\n"
                + "        return " + entityName.toLowerCase() + "List;\n"
                + "    }\n"
                + "}\n";
    }

}
