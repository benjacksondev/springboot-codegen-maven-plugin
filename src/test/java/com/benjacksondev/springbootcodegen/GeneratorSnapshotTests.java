package com.benjacksondev.springbootcodegen;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import com.benjacksondev.springbootcodegen.contentgenerators.ControllerGenerator;
import com.benjacksondev.springbootcodegen.contentgenerators.EntityGenerator;
import com.benjacksondev.springbootcodegen.contentgenerators.RepositoryGenerator;
import com.benjacksondev.springbootcodegen.contentgenerators.ServiceGenerator;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;

@ExtendWith({SnapshotExtension.class})
class GeneratorSnapshotTests {

    private Expect expect;

    @Test
    void generateControllerTest() {
        ControllerGenerator controllerGenerator = new ControllerGenerator();
        controllerGenerator.setComponentName("Gorilla");
        controllerGenerator.setPackageName("com.example.animals");

        ContentGeneratorResult result = controllerGenerator.generate();

        assertEquals("GorillaController", result.getClassName());
        expect.toMatchSnapshot(result.getClassContent());
    }

    @Test
    void generateServiceTest() {
        ServiceGenerator serviceGenerator = new ServiceGenerator();
        serviceGenerator.setComponentName("Zebra");
        serviceGenerator.setPackageName("com.example.animals");

        ContentGeneratorResult result = serviceGenerator.generate();

        assertEquals("ZebraService", result.getClassName());
        expect.toMatchSnapshot(result.getClassContent());
    }

    @Test
    void generateRepositoryTest() {
        RepositoryGenerator repositoryGenerator = new RepositoryGenerator();
        repositoryGenerator.setComponentName("Hippo");
        repositoryGenerator.setPackageName("com.example.animals");

        ContentGeneratorResult result = repositoryGenerator.generate();

        assertEquals("HippoRepository", result.getClassName());
        expect.toMatchSnapshot(result.getClassContent());
    }

    @Test
    void generateEntityTest() {
        EntityGenerator entityGenerator = new EntityGenerator();
        entityGenerator.setComponentName("Giraffe");
        entityGenerator.setPackageName("com.example.animals");

        ContentGeneratorResult result = entityGenerator.generate();

        assertEquals("Giraffe", result.getClassName());
        expect.toMatchSnapshot(result.getClassContent());
    }
    
}
