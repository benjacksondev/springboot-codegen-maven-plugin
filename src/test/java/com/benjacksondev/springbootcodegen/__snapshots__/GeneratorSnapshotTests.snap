com.benjacksondev.springbootcodegen.GeneratorSnapshotTests.generateControllerTest=[
package com.example.animals;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import jakarta.validation.Valid;

@Controller
@RequestMapping("/gorilla")
public class GorillaController {

    @Autowired
    GorillaService gorilla;

    @GetMapping("")
    public String showAll(@RequestParam(required = false) String message, Model model) {
        List<Gorilla> gorillas = gorilla.getGorillas();
        model.addAttribute("gorillas", gorillas);

        return "gorilla/index";
    }

    @PostMapping("/create")
    public String create(@Valid Gorilla gorilla, BindingResult result, Model model) {
        if (result.hasErrors()) {
            return "redirect:/gorilla/add";
        }

        gorillaservice.createGorilla(gorilla);
        model.addAttribute("message", "some message");
        return "redirect:/gorilla";
    }
}

]


com.benjacksondev.springbootcodegen.GeneratorSnapshotTests.generateEntityTest=[
package com.example.animals;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Giraffe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public Giraffe() {
    }

    public Long getId() {
        return id;
    }

}

]


com.benjacksondev.springbootcodegen.GeneratorSnapshotTests.generateRepositoryTest=[
package com.example.animals;

import org.springframework.data.repository.CrudRepository;

public interface HippoRepository extends CrudRepository<Hippo, Long> {
}

]


com.benjacksondev.springbootcodegen.GeneratorSnapshotTests.generateServiceTest=[
package com.example.animals;

import java.util.ArrayList;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ZebraService {

    @Autowired
    ZebraRepository zebraRepository;

    public Zebra createZebra(Zebra zebra) {
        return zebraRepository.save(zebra);
    }

    public List<Zebra> getZebras() {
        List<Zebra> zebraList = new ArrayList<>();
        zebraRepository.findAll().forEach(zebraList::add);
        return zebraList;
    }
}

]